cmake/kokkos_tpls.cmake

```diff
@@ -85,6 +85,17 @@ ENDIF()
 KOKKOS_IMPORT_TPL(ROCM INTERFACE)
 KOKKOS_IMPORT_TPL(LIBQUADMATH)
 
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
+  IF(KOKKOS_ENABLE_DESUL_ATOMICS_FORCE_EXTERNAL)
+    find_package(desul_atomics REQUIRED QUIET)
+  ELSE()
+    find_package(desul_atomics QUIET)
+  ENDIF()
+  IF(desul_atomics_FOUND)
+    KOKKOS_EXPORT_CMAKE_TPL(desul_atomics)
```

Comment:
It turned out that we don't properly export the `desul` dependency when using an external one.